command/bridge/finalize/finalize.go

```diff
@@ -103,6 +96,33 @@ func runCommand(cmd *cobra.Command, _ []string) error {
 	outputter := command.InitializeOutputter(cmd)
 	defer outputter.WriteOutput()
 
+	client, err := jsonrpc.NewEthClient(params.jsonRPC)
+	if err != nil {
+		return err
+	}
+
+	chainID, err := client.ChainID()
+	if err != nil {
+		return err
+	}
+
+	// get genesis config
+	chainConfig, err := chain.ImportFromFile(params.genesisPath)
+	if err != nil {
+		return fmt.Errorf("failed to read chain configuration: %w", err)
+	}
+
+	consensusConfig, err := polybft.GetPolyBFTConfig(chainConfig.Params)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve consensus configuration: %w", err)
+	}
+
+	bladeManagerAddr := consensusConfig.Bridge[chainID.Uint64()].BladeManagerAddr
+
+	if !consensusConfig.NativeTokenConfig.IsMintable {
+		consensusConfig.NativeTokenConfig.ChainID = chainID.Uint64()
```

Comment:
Why do you need to set this here? This is something the user needs to define manually in the `genesis` command, in the `native-token-config` (`nativeTokenConfigFlag`).

Remove this from here, and change `genesis` command accordingly. You would need to change the validation of the given flag. If the token is non mintable, the user needs to specify chain ID of origin.